web/modules/custom/dpl_paragraphs/dpl_paragraphs.module

```diff
@@ -108,3 +109,56 @@ function dpl_paragraphs_preprocess_paragraph__medias(array &$variables): void {
   $alternative_images = $paragraph->get('field_medias')->view('alternative');
   $variables['content']['field_medias'][1] = $alternative_images[1] ?? NULL;
 }
+
+/**
+ * Implements hook_preprocess_paragraph__TYPE().
+ *
+ * Prepares variables for paragraph--recommendation.html.twig template.
+ *
+ * If the paragraph has a title or description, we use the recommended-material
+ * React app. Otherwise, we use the recommendation React app.
+ */
+function dpl_paragraphs_preprocess_paragraph__recommendation(array &$variables): void {
+  /** @var \Drupal\paragraphs\ParagraphInterface $paragraph */
+  $paragraph = $variables['paragraph'];
+
+  if (!$paragraph->hasField('field_work_id') || $paragraph->get('field_work_id')->isEmpty()) {
+    return;
+  }
+  $work_id = $paragraph->get('field_work_id')->value;
```

Comment:
When possible, please use methods instead of directly accessing properties
e.g.

```suggestion
  $work_id = $paragraph->get('field_work_id')->getString();
```

(or `->getValue()`)

Reply:
same applies for the next 3 lines